pkg/datapath/loader/base.go

```diff
@@ -332,7 +352,7 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 
 	if option.Config.IPAM == ipamOption.IPAMENI {
 		var err error
-		if sysSettings, err = addENIRules(sysSettings); err != nil {
+		if sysSettings, err = addENIRules(sysSettings, nodeAddressing); err != nil {
```

Comment:
Do we have to pass `nodeAddressing`? Can't we get away with just calling `node.GetInternalIPv4Router()` inside `addENIRules` which would eliminate the need to change the interface of both `Reinitialize()` and `addENIRules()`.

Reply:
>Can't we get away with just calling node.GetInternalIPv4Router() inside

That's what I originally did, but decided to use the redirection via nodeAddressing in case we need to support IPv6 in this path (currently IPv6 doesn't seem to be supported with ENI IPAM mode though), and keep it consistent with the existing code.

Anyway I pushed a new commit couple of minutes back that gets rid off the additional argument.